python/tchannel/tornado/peer.py

```diff
@@ -188,11 +207,15 @@ def choose(self, hostport=None, score_threshold=None):
             If specified, Peers with a score equal to or below this will be
             ignored. Defaults to the value specified when the PeerGroup was
             initialized.
+        :param black_list:
+            Peers on the black_list won't be chosen.
         :returns:
             A Peer that matches all the requested criteria or None if no such
             Peer was found.
         """
-        if hostport:
+
+        black_list = black_list or []
+        if hostport and not self.get(hostport) in black_list:
```

Comment:
We're doing two `gets` here, does it make more sense for the blacklist to be a list of hostports instead of `Peers`?